control-plane/csi-controller/src/controller.rs

```diff
@@ -233,10 +234,23 @@ impl rpc::csi::controller_server::Controller for CsiControllerSvc {
             }
         };
 
+        // Check filesystem type.
+        if let Some(fs) = args.parameters.get("fsType") {
+            if !SUPPORTED_FS_TYPES.iter().any(|f| f == &fs.to_string()) {
+                return Err(Status::invalid_argument("Invalid filesystem type"));
+            }
+        };
+
         // Check storage protocol.
         let protocol = match args.parameters.get("protocol") {
-            Some(p) => p.to_string(),
-            None => return Err(Status::invalid_argument("Missing storage protocol")),
+            Some(p) => {
```

Comment:
We already have a `parse_protocol` which is used on publish and still accepts icsi there.
Instead why not call that function here, maybe change it to receive an option an reuse it so the logic will be in a single function, eg:
```rust
let protocol = parse_protocol(args.parameters.get("protocol"))?;
```